inc/classes/inserter/wp/attachment.php

```diff
@@ -13,42 +13,40 @@ class Attachment extends Post {
 	/**
 	 * Upload and add attachment object into the database
 	 *
-	 * @param array $path
-	 * @param array $post_data
-	 * @param bool $canonical_id
-	 * @param array $post_meta
-	 * @param null $file_type_override
-	 * @param bool $force_update_existing
+	 *
+	 * @param array $post_data    Post data formatted as it will be saved to the posts table. Should match WP_Post data.
+	 * @param bool  $canonical_id Use an existing canonical ID.
+	 * @param array $post_meta    Metadata to assign to the post.
+	 * @param array $options      Additional data about the post.
 	 * @return array|int|object|\WP_Error
 	 */
-	static function insert( $path, $post_data = array(), $canonical_id = false, $post_meta = array(), $file_type_override = null, $force_update_existing = true ) {
-
+	static function insert( $post_data = [], $canonical_id = false, $post_meta = [], $options = [] ) {
```

Comment:
It majorly sucks that the most prominent arg has to be defined as an element of an array in the last function arg.

Probably best way around this for now would be

```
function insert_from_path( $path, $post_data = [], $canonical_id = false, $post_meta = [], $options = []  ) {
    return static::insert( $post_data ...
}
```

```
function insert_from_url( $url, $post_data = [], $canonical_id = false, $post_meta = [], $options = []  ) {
    return static::insert( $post_data ...
}
```